### PR TITLE
feat: Add result bundle options as part of capabilities

### PR DESCRIPTION
**Pull request by KazuCocoa**
https://github.com/appium/WebDriverAgent/pull/410 for xcuitest driver.

Must bump the module version after the 410.

Example:
```
caps = {
  platformName: :xcuitest,
  ...
  resultBundlePath: "/Users/kazu/GitHub/ruby_lib_core/tmp/#{Time.now.strftime('%Y-%m-%d-%H%M%S')}",
  resultBundleVersion: 3,
  useNewWDA: true  # To separate `xcresult`, the client must stop WDA process everytime. So far, this useNewWDA avoid using running WDA on the device under test.
}
driver = ::Appium::Core.for(caps).start_driver
// do scenario
```

Then, like below files will be created in the result bundle path.

```
/Users/kazu/GitHub/ruby_lib_core/tmp/2020-10-24-112035 # results will be in this directory
                                     2020-10-24-112035.xcresult
```

Then, `2020-10-24-112035.xcresult` is able to open by Xcode (or `xcresulttool`)

---

When we open the `.xcresult` by Xcode, then below result appears. (Xcode 12)
image

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
Could you please explain how these two capabilities could be used? Just an example of a client script

**Comment by KazuCocoa** [comment]
I've updated the description.
This PR just add two capabilities, but probably need a followup PR to add a capability to stop WDA process?

**Comment by KazuCocoa** [comment]
Will update unit test stuff after https://github.com/appium/appium-xcuitest-driver/pull/1248

**Comment by KazuCocoa** [comment]
Updated

File: README.md [hunk 1]
```diff
@@ -132,6 +132,8 @@ Differences are noted here:
 |`processArguments`|Process arguments and environment which will be sent to the `WebDriverAgent` server.|`{ args: ["a", "b", "c"] , env: { "a": "b", "c": "d" } }` or `'{"args": ["a", "b", "c"], "env": { "a": "b", "c": "d" }}'`|
 |`autoLaunch`|When set to `false`, prevents the application under test from being launched automatically as a part of the new session startup process. The launch become the responsibility of the user. Defaults to `true`.|`true` or `false`|
 |`allowProvisioningDeviceRegistration`|Allow `xcodebuild` to register your destination device on the developer portal if necessary. Requires a developer account to have been added in Xcode's Accounts preference pane. Defaults to `false`.|`true` or `false`|
+|`resultBundlePath`| Specify the path to the result bundle path as `xcodebuild` argument for `WebDriverAgent` build. `WebDriverAgent` process must start/stop every time to set this path properly. `useNewWDA` may help then. Please read `xcodebuild` command help for more details. | e.g. `/path/to/resultbundle` |
```
Comment (mykola-mokhnach) [inline README.md:135]: `to set this path properly` - I would rather say `to pick up changed value of this property`

Comment (mykola-mokhnach) [inline README.md:135]: Setting `useNewWDA` to `true` may help there

Comment (mykola-mokhnach) [reply README.md:135]: Please read `man xcodebuild` for more details.

Comment (mykola-mokhnach) [inline README.md:135]: From security perspective it makes sense to restrict the root path, for example to user home folder

Comment (KazuCocoa) [reply README.md:135]: ah, that's right. Let me add a security flag

File: lib/driver.js [hunk 2]
```diff
@@ -541,6 +543,11 @@ class XCUITestDriver extends BaseDriver {
         throw new Error(msg);
       };
 
+      // Used in following WDA build
```
Comment (mykola-mokhnach) [inline lib/driver.js:546]: in the following

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
We probably would need to bump WDA after https://github.com/appium/WebDriverAgent/pull/415 is merged to fix the tests

**Comment by KazuCocoa** [comment]
mm. Will check failing tests later

**Comment by KazuCocoa** [comment]
I checked failed tests, Integration_Tests e2e_basic_iPhoneX_13_7, on my local, but they worked

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-xcuitest-driver/pull/1252 has the same CI failures.
Will take a look at it as another PR to fix CI or WDA? (need to dig them more)

**Pull request merged**